File: app/src/main/java/deltazero/amarok/filehider/NoneFileHider.java
package deltazero.amarok.filehider;

import android.content.Context;

import java.util.Set;

public class NoneFileHider extends BaseFileHider {

    public NoneFileHider(Context context) {
        super(context);
    }

    @Override
    protected void process(Set<String> targetDirs, ProcessMethod method) throws InterruptedException {
        return;
    }

    @Override
    public void tryToActive(ActivationCallbackListener activationCallbackListener) {
        activationCallbackListener.onActivateCallback(this.getClass(), true, 0);
    }

    @Override
    public String getName() {
        return "Disabled";
    }
}

File: app/src/main/java/deltazero/amarok/filehider/NoneFileHider.kt
package deltazero.amarok.filehider

import deltazero.amarok.core.ActivationResult
import deltazero.amarok.core.Hider

class NoneFileHider : FileHider {
  override val mode = FileHiderMode.NONE
  override val name = "Disabled"

  override suspend fun activate() = ActivationResult(success = true, msgResId = 0)

  override suspend fun process(targetDirs: Set<String>, action: Hider.Action) {
    // No-op
  }
}
